pysisu/pysisu_class.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright 2022 Sisu Data, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from io import UnsupportedOperation
+from typing import Union
+import requests
+from pysisu.formats import LatestAnalysisResultsFormats, Table
+from pysisu.latest_analysis_result import to_table
+from pysisu.query_helpers import build_url, pathjoin
+from pysisu.sisu.v1.api import LatestAnalysisResultResponse
+
+
+class PySisu:
+    '''
+    Allows the ability to fetch/send commands to the sisu_api.
+    '''
+
+    _url: str
+    _api_key: str
+
+    def __init__(self, api_key: str, url: str = 'https://vip.sisudata.com') -> None:
+        self._url = url
+        self._api_key = api_key
+
+    def set_url(self, url: str) -> "PySisu":
+        self._url = url
+        return self
+
+    def set_api_key(self, api_key: str) -> "PySisu":
+        self._api_key = api_key
+        return self
+
+    def auto_paginate(
+        self,
+        analysis_id: int,
+        params: dict,
+        result: LatestAnalysisResultResponse,
+    ) -> None:
+        kda_result = result.analysis_result.key_driver_analysis_result
+        subgroups = kda_result.subgroups
+        if subgroups:
+            return result
+        if params.get('limit'):
+            params['limit'] -= len(subgroups)
+        if params.get('limit', 0) < 0:
+            return result
+
+        rest_of_table = self.get_results(
```

Comment:
```suggestion
        next_page = self.get_results(
```
this is a recursive call, rest_of_table is partial and might be good to make it more obvious through naming that it's recursive